main.py

```diff
@@ -419,23 +431,30 @@ def gcs_to_pubsub(cloud_event: CloudEvent):
     Returns:
         None
     """
-    logging.info("Cloud event triggered")
+    logging.debug("Cloud event triggered")
 
     data = cloud_event.data
+
+    # Skip processing if filename starts with "temp"
+    if "/temp_" in data["name"]:
+        logging.warning("Blob name contains '/temp_', skipping processing")
+        return
+
+    # Skip processing folders
+    if data["name"][-1] == "/":
```

Comment:
is there not a type attribute of some kind to differentiate between files and folders?

Reply:
Sadly no, we tried looking to exit sooner and this was the earliest